### PR TITLE
Add a test to verify no more use of deprecated env syntax

### LINKED ISSUE #538: kustomize syntax "env: params.env" deprecated in next release of kustomize; kubeflow stuck on kustomize 3.2.1
Heads up, just learned the hard way:

the following syntax is not supported anymore (since 10/15/2019)

```yaml
configMapGenerator:
- env: params.env
  name: parameters
````

It needs to be:

```yaml
configMapGenerator:
- envs:
  - params.env
  name: parameters
```

Will have to enforce that the user uses a very specific version of kustomize until the manifests are updated.

### PR DESCRIPTION
**Pull request by jlewi**
* Fix one remaining use in fluentd

* Fix #538

### CODE REVIEW

**Comment by jlewi** [comment]
Looks like the test failure is unrelated to this PR; looks like a problem monitoring the Argo workflow.

```
INFO|2020-11-05T16:55:34|/src/kubeflow/testing/py/kubeflow/testing/argo_client.py|60| ApiException code=404
INFO|2020-11-05T16:55:44|/src/kubeflow/testing/py/kubeflow/testing/argo_client.py|48| Parsing ApiException body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"workflows.argoproj.io \"kubeflow-manifests-presubmit-e2e-unit-1610-2c1af4b-4432-a44e\" not found","reason":"NotFound","details":{"name":"kubeflow-manifests-presubmit-e2e-unit-1610-2c1af4b-4432-a44e","group":"argoproj.io","kind":"workflows"},"code":404}
INFO|2020-11-05T16:55:44|/src/kubeflow/testing/py/kubeflow/testing/argo_client.py|60| ApiException code=404
ERROR|2020-11-05T16:55:44|/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py|528| Exception occurred: ApiException()
Traceback (most recent call last):
  File "/src/kubeflow/testing/py/kubeflow/testing/argo_client.py", line 136, in wait_for_workflows
    results = get_namespaced_custom_object_with_retries(namespace, name)
  File "/usr/local/lib/python3.8/dist-packages/retrying.py", line 49, in wrapped_f
    return Retrying(*dargs, **dkw).call(f, *args, **kw)
  File "/usr/local/lib/python3.8/dist-packages/retrying.py", line 212, in call
    raise attempt.get()
  File "/usr/local/lib/python3.8/dist-packages/retrying.py", line 247, in get
    six.reraise(self.value[0], self.value[1], self.value[2])
  File "/usr/lib/python3/dist-packages/six.py", line 693, in reraise
    raise value
  File "/usr/local/lib/python3.8/dist-packages/retrying.py", line 200, in call
    attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
  File "/src/kubeflow/testing/py/kubeflow/testing/argo_client.py", line 104, in get_namespaced_custom_object_with_retries
    return crd_api.get_namespaced_custom_object(
  File "/usr/local/lib/python3.8/dist-packages/kubernetes/client/apis/custom_objects_api.py", line 931, in get_namespaced_custom_object
    (data) = self.get_namespaced_custom_object_with_http_info(group, version, namespace, plural, name, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/kubernetes/client/apis/custom_objects_api.py", line 1018, in get_namespaced_custom_object_with_http_info
    return self.api_client.call_api('/apis/{group}/{version}/namespaces/{namespace}/{plural}/{name}', 'GET',
  File "/usr/local/lib/python3.8/dist-packages/kubernetes/client/api_client.py", line 330, in call_api
    return self.__call_api(resource_path, method,
  File "/usr/local/lib/python3.8/dist-packages/kubernetes/client/api_client.py", line 163, in __call_api
    response_data = self.request(method, url,
  File "/usr/local/lib/python3.8/dist-packages/kubernetes/client/api_client.py", line 351, in request
    return self.rest_client.GET(url,
  File "/usr/local/lib/python3.8/dist-packages/kubernetes/client/rest.py", line 227, in GET
    return self.request("GET", url,
  File "/usr/local/lib/python3.8/dist-packages/kubernetes/client/rest.py", line 222, in request
    raise ApiException(http_resp=r)
kubernetes.client.rest.ApiException: (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Audit-Id': 'fbbebdf7-91b2-48ad-b476-409ece7ee804', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'Date': 'Thu, 05 Nov 2020 16:55:44 GMT', 'Content-Length': '332'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"workflows.argoproj.io \"kubeflow-manifests-presubmit-e2e-unit-1610-2c1af4b-4432-a44e\" not found","reason":"NotFound","details":{"name":"kubeflow-manifests-presubmit-e2e-unit-1610-2c1af4b-4432-a44e","group":"argoproj.io","kind":"workflows"},"code":404}
```

File: aws/fluentd-cloud-watch/base_v3/kustomization.yaml [hunk 1]
```diff
@@ -17,7 +17,8 @@ images:
   newTag: v1.7.3-debian-cloudwatch-1.0
 configMapGenerator:
 - name: fluentd-cloud-watch-parameters
-  env: params.env
+  envs:
```
Comment (Jeffwan) [inline aws/fluentd-cloud-watch/base_v3/kustomization.yaml:20]: nice catch. Thanks!

**Review by Jeffwan — APPROVED** [review]
/lgtm

**Comment by Jeffwan** [comment]
Seems it tries to find  `86308603-argo-argo-5ce9-1162466691.us-west-2.elb.amazonaws.com/workflows/kubeflow-test-infra/kubeflow-manifests-presubmit-e2e-unit-1610-2c1af4b-4432-a44e?tab=workflow` However, the workflow can not be find.  

@PatrickXYS Can you help check if it's naming issue?

**Comment by PatrickXYS** [comment]
/test kubeflow-manifests-presubmit-e2e

@Jeffwan By default, the Argo Workflow TTL is set to be 1hour, then GC.

Basically I think it's one-off, let's try again

**Comment by Jeffwan** [comment]
@PatrickXYS  This workflow takes over 1 hour to finish. We may need to extend TTL. EKS creation and code fetching is kind of slow. You can check EFS performance data and probably change burst model later. Don't forget to check estimate cost

**Comment by PatrickXYS** [comment]
Argo Workflow TTL (ttlSecondsAfterFinished) is the parameter to indict the remaining time **after** workflow finished.

So argo workflow can take 24h to finish, around 25h, the workflow will be GC

**Comment by PatrickXYS** [comment]
I checked EFS throughput and IO performance, looks pretty normal and has not exceed any limitation. 

Maybe I need to do some experiments

**Comment by Jeffwan** [comment]
> Argo Workflow TTL (ttlSecondsAfterFinished) is the parameter to indict the remaining time **after** workflow finished.
> 
> So argo workflow can take 24h to finish, around 25h, the workflow will be GC

My point is unit workflow will be created immediately and finished probably at 5mins. Then the other one runs for > 57mins. At this time, it tried to fetch workflow. However, unit-workflow has been recycled since 5 + 57 = 62 mins > 1hr. 

Extending TTL makes sure the one finish quicker won't be recycled if there's two workflows running at the same time. 
In this case, unit + 1hr < e2e running time.

**Comment by Jeffwan** [comment]
@PatrickXYS Let's make some updates on the configuration

**Comment by PatrickXYS** [comment]
@Jeffwan Yeah that makes sense,

PR to increase unit test TTL https://github.com/kubeflow/manifests/pull/1611

**Pull request merged**